### PR TITLE
use context methods instead of yield in hacker news jobs

### PR DESCRIPTION
**Pull request by sryza**
## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by dpeng817 — APPROVED** [review]
LGTM!

File: examples/hacker_news/hacker_news/ops/download_items.py [hunk 1]
```diff
@@ -53,12 +47,12 @@ def download_items(context, id_range: Tuple[int, int]) -> Output:
     result = DataFrame(non_none_rows, columns=ACTION_FIELD_NAMES).drop_duplicates(subset=["id"])
     result.rename(columns={"by": "user_id"}, inplace=True)
 
-    return Output(
-        result,
-        "items",
-        metadata={"Non-empty items": len(non_none_rows), "Empty items": rows.count(None)},
+    context.add_output_metadata(
```
Comment (schrockn) [inline examples/hacker_news/hacker_news/ops/download_items.py:50]: Is this really superior? I guess the typehint feels better.

**Comment by schrockn** [comment]
Honestly I think this mostly a regression. `return Output` IMO is better

**Comment by schrockn** [comment]
It just feels much less functional and testable.

**Comment by sryza** [comment]
Hmm, I think you're right that we might be throwing the baby (`Output`) out with the bathwater (`yield Output`) here. We could likely get the best of both worlds if we make `Output` a generic type.

E.g.
```
@asset
def items() -> Output[DataFrame]:
    ...
```

**Comment by sryza** [comment]
Closing in favor of https://github.com/dagster-io/dagster/pull/7600

**Pull request closed without merging**